sync/worker_pool_test.go

```diff
@@ -121,3 +121,23 @@ func TestGoWithTimeout(t *testing.T) {
 
 	require.Equal(t, uint32(testWorkerPoolSize+1), count)
 }
+
+func TestInitOnInitializedWorkerPool(t *testing.T) {
+	var count uint32
```

Comment:
Feels a better test would be:
```
p := NewWorkerPool(testWorkerPoolSize)
	p.Init()

	var wg sync.WaitGroup
	wg.Add(1)
	go func() {
	  p.Init()
	  wg.Done()
	}
	wg.Wait()
```
This way it's clear that it's avoiding the hanging, rather than checking the counts and stuff

Reply:
Sure. I was thinking there should be some assert in the test but I suppose that is not necessarily true.